File: 003_sum_product/program.nim

#import strformat

# #[ this was multile comment
var sum: int
echo "Sum numbers 1-100:"

for num in 1 .. 100 :
  sum = sum + num

echo sum
# ]#

var prod: int64 = 1
echo "Product numbers 1-15:"

for num2 in 1'i64 .. 15'i64 :
  prod = prod * num2

echo prod

File: 003_sum_product/program.zig

const std = @import("std");
const print = std.debug.print;

pub fn main() !void{
  var sum: u16 = 0;
  for(1..101) |i|{
    sum += @intCast(i);
  }
  
  print("Sum numbers 1-100:\n{d}\n", .{sum});
  
  var prod: u64 = 1;
  for(1..16) |i|{
    prod *= i;
  }
  print("Product numbers 1-15:\n{d}\n", .{prod});
}

